src/main/diagnostics/diagnostics.ts

```diff
@@ -82,10 +85,32 @@ export class DiagnosticsManager {
 
   constructor(diagnostics?: DiagnosticsChecker[]) {
     this.checkers = diagnostics ? Promise.resolve(diagnostics) : (async() => {
-      return (await Promise.all([
+      const checkers = (await Promise.all([
         import('./connectedToInternet'),
         import('./dockerCliSymlinks'),
-      ])).map(obj => obj.default).filter(checker => checker.applicable);
+        import('./rdBinInShell'),
+      ])).flatMap(obj => obj.default);
+
+      // Check if the checkers are applicable; this may require waiting on
+      // dependent checkers.
+      const applicablePromise: Record<string, Promise<boolean>> = {};
+      const setup = Latch();
+
+      for (const checker of checkers) {
```

Comment:
I found this block of new code confusing. Why do we need `DiagnosticsChecker.applicable` to take a function as an argument? I see that with the new code from `rdBinInShell.ts` there is a need that `DiagnosticsChecker.applicable` can be async, but I don't see the need for passing a function in if we simplify this code. Also, what is the purpose of the `setup` latch? I don't see it referenced anywhere else?

Why not make this block of code something like:
```
      const filteredCheckers = [];
      for (const checker of checkers) {
        if (await checker.isApplicable()) {
          filteredCheckers.push(checker);
        }
      }
      return filteredCheckers;
```
It doesn't take advantage of async, but I don't think it matters here?

Reply:
Dropped the complication, now it's a `() => Promise<boolean>` instead.
(The extra code would be useful once we have checkers that are only useful if a different checker passes, but I think that was implemented wrong anyway, so we'll need to handle that differently in a future PR.)
Fixed in c344f47d400c7690d71d398c574302a3a084359e.